lib/nats.js

```diff
@@ -1670,6 +1813,15 @@ Client.prototype.request = function(subject, opt_msg, opt_options, callback) {
         opt_options = null;
     }
 
+    if (this.draining) {
+        if (typeof callback === 'function') {
```

Comment:
This code seems duplicated alot, maybe make a helper function?

Reply:
fixed.